scripts/bench_python_cpp_algos.py

```diff
@@ -353,73 +365,67 @@ def load_graph(graph_path, edge_properties=None):
+        "sssp": (sssp, (args.source_nodes)),
+        "jaccard": (jaccard, ()),
+        "bc": (bc, (args.source_nodes, args.num_sources)),
+        "tc": (tc, ()),
+        "cc": (cc, ()),
+        "kcore": (kcore, ()),
+        "louvain": (louvain, ()),
+        "pagerank": (pagerank, ()),
+    }
 
-    elif args.application in ["tc"]:
-        graph_path = f"{args.input_dir}/{input['symmetric_clean_input']}"
-        if not os.path.exists(graph_path):
-            print(f"Symmetric clean Graph doesn't exist: {graph_path}")
+    if args.application not in ["all"]:
+        if args.application in ["bfs", "sssp", "bc", "jaccard"]:
+            graph_path = f"{args.input_dir}/{input['name']}"
+            if not os.path.exists(graph_path):
+                print(f"Graph doesn't exist: {graph_path}")
 
-        graph = load_graph(graph_path, [])
+            graph = load_graph(graph_path)
 
-        if args.application == "tc":
-            data = run_routine(run_tc, data, args.trials, (graph, input))
+        elif args.application in ["tc"]:
+            graph_path = f"{args.input_dir}/{input['symmetric_clean_input']}"
+            if not os.path.exists(graph_path):
+                print(f"Symmetric clean Graph doesn't exist: {graph_path}")
 
-    elif args.application in ["cc", "kcore"]:
-        graph_path = f"{args.input_dir}/{input['symmetric_input']}"
-        if not os.path.exists(graph_path):
-            print(f"Symmetric Graph doesn't exist: {graph_path}")
+            graph = load_graph(graph_path, [])
 
-        graph = load_graph(graph_path, [])
+        elif args.application in ["cc", "kcore"]:
+            graph_path = f"{args.input_dir}/{input['symmetric_input']}"
+            if not os.path.exists(graph_path):
+                print(f"Symmetric Graph doesn't exist: {graph_path}")
 
-        if args.application == "cc":
-            data = run_routine(run_cc, data, args.trials, (graph, input))
+            graph = load_graph(graph_path, [])
 
-        if args.application == "kcore":
-            data = run_routine(run_kcore, data, args.trials, (graph, input))
+        elif args.application in ["louvain"]:
+            graph_path = f"{args.input_dir}/{input['symmetric_input']}"
+            if not os.path.exists(graph_path):
+                print(f"Symmetric Graph doesn't exist: {graph_path}")
 
-    elif args.application in ["louvain"]:
-        graph_path = f"{args.input_dir}/{input['symmetric_input']}"
-        if not os.path.exists(graph_path):
-            print(f"Symmetric Graph doesn't exist: {graph_path}")
+            graph = load_graph(graph_path)
 
-        graph = load_graph(graph_path)
+        elif args.application in ["pagerank"]:
+            # Using transpose file pagerank pull which is expected
+            # to perform better than pagerank push algorithm
+            graph_path = f"{args.input_dir}/{input['transpose_input']}"
+            if not os.path.exists(graph_path):
```

Comment:
Maybe extract all this if statements that check if graph exists at the end of the outer if/else; then you need to check only once